src/core/sparsemat.c

```diff
@@ -700,11 +700,18 @@ int igraph_sparsemat_fkeep(igraph_sparsemat_t *A,
                            int (*fkeep)(int, int, igraph_real_t, void*),
                            void *other) {
 
-    if (!cs_fkeep(A->cs, fkeep, other)) {
-        IGRAPH_ERROR("Cannot filter sparse matrix", IGRAPH_FAILURE);
+    if (!A) {
```

Comment:
I think we can skip this check, as a null pointer wouldn't purposefully be passed in here.

Most of the checks / error reports are for conditions that can be triggered from a high-level interface. Passing in a null pointer cannot.

This situation would be comparable to checking every `igraph_t *graph` not to be null, which would IMO be too much.

Reply:
The specific reason I put this here is because cs_fkeep checks for it. If you don't check it here, cs_fkeep would return -1, and it's strange not to handle that error. And if you're going to handle it anyway, this is clearer.

Reply:
Issuing a proper error for this still bothers me ... I am not against checking, I am against `IGRAPH_ERROR`.  I know that this is subjective, but `IGRAPH_ERROR` generally indicates "expected errors" while `IGRAPH_ASSERT` indicates "unexpected errors".

Error messages should generally be understandable to a user of the R/Python/Mathematica interfaces, or a hypothetical GUI program using igraph. They could be triggered by something the user did. In that context "null pointer" makes no sense—what  can a user do to avoid it? If it happened, it's the fault of the programmer, not the fault of the end user. It's a bug that should be fixed in the software that depend on igraph, not a mistake made by the user of that software.

Therefore, I would go for an `IGRAPH_ASSERT` instead.